### PR TITLE
pretty_symbology: Remove Python 2 fallbacks

### PR DESCRIPTION
**Pull request by eric-wieser**
`unicodedata` was added in Python 2.5. Since Python 2.4 is very not supported, we can get rid of this

This also removes a section of code marked for 2.X removal, and cleans up some exception handling

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/printing/pretty/pretty_symbology.py [hunk 1]
```diff
@@ -73,33 +58,32 @@ def pretty_use_unicode(flag=None):
...
+        encoding = getattr(sys.stdout, 'encoding', None)
 
-            # this happens when e.g. stdout is redirected through a pipe, or is
-            # e.g. a cStringIO.StringO
-            if encoding is None:
-                return  # sys.stdout has no encoding
+        # this happens when e.g. stdout is redirected through a pipe, or is
+        # e.g. a cStringIO.StringO
+        if encoding is None:
+            return  # sys.stdout has no encoding
```
Comment (oscarbenjamin) [inline sympy/printing/pretty/pretty_symbology.py:78]: I guess this could be at the top of the function.

**Comment by oscarbenjamin** [comment]
Looks good to me

File: sympy/printing/pretty/pretty_symbology.py (hunk 1 above)
Comment (eric-wieser) [reply sympy/printing/pretty/pretty_symbology.py:78]: Oh yeah, doing this in the loop is pretty silly.

Comment (eric-wieser) [reply sympy/printing/pretty/pretty_symbology.py]: Done

**Comment by oscarbenjamin** [comment]
Closing and reopening to restart the tests.

**Pull request merged**